tests/Unit/IO/Test_VolumeData.py

```diff
@@ -15,25 +15,88 @@ class TestIOH5VolumeData(unittest.TestCase):
-        # EvolvePlaneWave3D executable.  However, the file does not contain the
-        # source tree (for file size reasons), so it should not be used as an
-        # example output for any other reason than testing VolumeData.
-        self.file_name_r = os.path.join(Informer.unit_test_path(),
-                                        "IO/VolTestData.h5")
-        if os.path.isfile(self.file_name_w):
-            os.remove(self.file_name_w)
+        # The other tests use a volume data file written using
+        # the write_volume_data() function
+        self.file_name = os.path.join(Informer.unit_test_path(),
+                                      "IO/pythontest.h5")
+
+        if os.path.isfile(self.file_name):
+            os.remove(self.file_name)
+
+        h5_file = spectre_h5.H5File(file_name=self.file_name,
+                                    append_to_file=True)
+        self.tensor_components_and_coords = [
+            ds.DataVector([8.9, 7.6, 3.9, 2.1, 18.9, 17.6, 13.9, 12.1]),
+            ds.DataVector([0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0]),
+            ds.DataVector([0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0]),
+            ds.DataVector([0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0]),
+            ds.DataVector([-78.9, -7.6, -1.9, 1.9, 6.3, 1.7, 9.8, 1.2]),
+            ds.DataVector([-7.9, 17.6, 1.9, -8.1, -6.3, 21.7, 6.8, -0.2]),
+            ds.DataVector([1.0, 6.6, 28.0, -8.1, -26.3, 32.7, 26.8, -3.2])
+        ]
+        observation_ids = [8435087234, 6785087234]
+        observation_values = [7.0, 1.3]
+        grid_names = ["[[2,3,4]]", "[[5,6,7]]"]
+
+        # Insert .vol file to h5 file
+        h5_file.insert_vol("/element_data", 0)
+        vol_file = h5_file.get_vol(path="/element_data")
+        # Set TensorComponent and ExtentsAndTensorVolumeData to
+        # be written
+        extents_and_tensor_vol_obs1 = [
+            ds.ExtentsAndTensorVolumeData([2, 2, 2], [
+                ds.TensorComponent(grid_names[0] + "/S",
+                                   self.tensor_components_and_coords[0]),
+                ds.TensorComponent(grid_names[0] + "/x-coord",
+                                   self.tensor_components_and_coords[1]),
+                ds.TensorComponent(grid_names[0] + "/y-coord",
+                                   self.tensor_components_and_coords[2]),
+                ds.TensorComponent(grid_names[0] + "/z-coord",
+                                   self.tensor_components_and_coords[3]),
+                ds.TensorComponent(grid_names[0] + "/T_x",
+                                   self.tensor_components_and_coords[4]),
+                ds.TensorComponent(grid_names[0] + "/T_y",
+                                   self.tensor_components_and_coords[5]),
+                ds.TensorComponent(grid_names[0] + "/T_z",
+                                   self.tensor_components_and_coords[6])
+            ]) for observation_value in observation_values
+        ]
```

Comment:
I was confused by the suffix `_obs1`, I wrongly thought the data corresponds to the first observation and the `_obs2` data corresponds to the second observation. Please name `_grid1` and `_grid2` instead.

You don't need these 7 fields here, I think 2 are enough, and you can swap their order for the two observations (that's why I suggested to generate just 4 rows of random data in the comment above). You can call them `field1` and `field2`, for example.

Reply:
Yes, sorry, that should be grids, and will reduce the number of fields.